front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -0,0 +1,196 @@
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'stopFor'>(
+          'stopFor',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+          })
+        ),
+        title: `${t('stopTime')}`,
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'onStopSignal'>(
+          'onStopSignal',
+          checkboxColumn as Partial<Column<boolean | undefined>>
+        ),
+        title: t('receptionOnClosedSignal'),
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'theoreticalMargin'>(
+          'theoreticalMargin',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+            placeholder: t('theoreticalMarginPlaceholder'),
+            formatBlurredInput: (value) => {
+              if (!value || value === 'none') return '';
+              if (!marginRegExValidation.test(value)) {
+                return `${value}${t('theoreticalMarginPlaceholder')}`;
+              }
+              return value;
+            },
+          })
+        ),
+        cellClassName: ({ rowData }) => (rowData.isMarginValid ? '' : 'invalidCell'),
+        title: t('theoreticalMargin'),
+        disabled: ({ rowIndex }) => rowIndex === findLastIndex(pathProperties.allVias),
+      },
+    ],
+    [t, pathProperties, timesStopsSteps]
+  );
+
+  const removeVia = (
+    rowData: PathWaypointColumn,
+    setRowData: (rowData: PathWaypointColumn) => void
+  ) => {
+    const updatedPathSteps = [...pathSteps];
+    const index = compact(updatedPathSteps).findIndex((step) => {
+      if ('uic' in step) {
+        return step.uic === rowData.uic && step.ch === rowData.ch && step.kp === rowData.kp;
+      }
+      if ('track' in step) {
+        return step.track === rowData.track;
+      }
+      return false;
+    });
+    updatedPathSteps.splice(index, 1);
+
+    setRowData({
+      ...rowData,
+      arrival: undefined,
+      stopFor: undefined,
+      theoreticalMargin: undefined,
+      onStopSignal: false,
+    });
+    dispatch(updatePathSteps(updatedPathSteps));
+  };
+
+  return (
+    <DynamicDataSheetGrid
+      columns={columns}
+      value={timesStopsSteps}
+      onChange={(e, [op]) => {
+        const rowData = e[`${op.fromRowIndex}`];
+        if (rowData.theoreticalMargin && !marginRegExValidation.test(rowData.theoreticalMargin!)) {
+          rowData.isMarginValid = false;
+        } else {
+          rowData.isMarginValid = true;
+          dispatch(upsertViaFromSuggestedOP(rowData as SuggestedOP));
+          setPathProperties({ ...pathProperties, allVias: e as SuggestedOP[] });
+        }
+        setTimesStopsSteps(e as PathWaypointColumn[]);
+      }}
+      stickyRightColumn={{
+        component: ({ rowData, setRowData }) =>
+          deleteViaButton(
+            () => removeVia(rowData as PathWaypointColumn, setRowData),
```

Comment:
I think you can remove all the cast l179, 180, 188 and 172
```suggestion
            () => removeVia(rowData, setRowData),
```

Reply:
Done